src/cargo/util/network/sleep.rs

```diff
@@ -0,0 +1,94 @@
+//! Utility for tracking network requests that will be retried in the future.
+
+use core::cmp::Ordering;
+use std::collections::BinaryHeap;
+use std::time::{Duration, Instant};
+
+pub struct SleepTracker<T> {
+    heap: BinaryHeap<Sleeper<T>>,
```

Comment:
Does this need to be a min heap?

Reply:
That all seams good. Should we have a comment on the `BinaryHeap` and the `PartialEq` pointing out this important detail?